### ISSUE TITLE
Errors Middleware + IngressRoute : Could not get Capture / value not found in context

### ISSUE THREAD

**Issue by guyguy333** [traefik/traefik#10820]
### Welcome!

- [X] Yes, I've searched similar issues on [GitHub](https://github.com/traefik/traefik/issues) and didn't find any.
- [X] Yes, I've searched similar issues on the [Traefik community forum](https://community.traefik.io) and didn't find any.

### What did you do?

I did enable middlewares on dashboard ingress route as available in Helm Chart :

https://github.com/traefik/traefik-helm-chart/blob/v28.3.0/traefik/values.yaml#L167

Something like:

```
        middlewares:
          - name: traefik-errors
            namespace: ingress-controller
          - name: traefik-forwardauth
            namespace: ingress-controller
```

However, every time I reach dashboard, I've a white page.

### What did you see instead?

A white page and an error log instead of custom error page : 

```json
{
  "level": "debug",
  "middlewareName": "ingress-controller-traefik-forwardauth@kubernetescrd",
  "middlewareType": "CustomError",
  "time": "2024-06-17T19:08:38Z",
  "caller": "github.com/traefik/traefik/v3/pkg/middlewares/customerrors/custom_errors.go:87",
  "message": "Caught HTTP Status Code 401, returning error page"
}
{
  "level": "error",
  "service": "ingress-controller-traefik-errors-errorpage-service@kubernetescrd",
  "error": "value not found in context",
  "time": "2024-06-17T19:08:38Z",
  "caller": "github.com/traefik/traefik/v3/pkg/middlewares/metrics/metrics.go:147",
  "message": "Could not get Capture"
}
```

Disabling errors middleware, produces the expected result. It looks like issue is related to errors middleware.

Also, If I setup same middlewares on a Kubernetes Ingress with same domain and with a service redirecting to traefik port, I don't have the issue, everything is working as expected.

```yaml
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: traefik-dashboard
  namespace: ingress-controller
  annotations:
    traefik.ingress.kubernetes.io/router.middlewares: ingress-controller-traefik-forwardauth@kubernetescrd,ingress-controller-traefik-errors@kubernetescrd
spec:
  ingressClassName: traefik
  tls:
    - hosts:
        - example.com
      secretName: example.com
  rules:
    - host: example.com
      http:
        paths:
          - path: /
            pathType: Prefix
            backend:
              service:
                name: traefik-traefik
                port:
                  number: 9000
---
apiVersion: v1
kind: Service
metadata:
  name: traefik-dashboard
  namespace: ingress-controller
spec:
  ports:
    - name: traefik
      protocol: TCP
      port: 9000
      targetPort: traefik
  selector:
    app.kubernetes.io/instance: traefik-ingress-controller
    app.kubernetes.io/name: traefik
  type: ClusterIP
  internalTrafficPolicy: Cluster
```

### What version of Traefik are you using?

3.0.2

### What is your environment & configuration?

```yaml
# See middleware config above
```

Helm Chart

### If applicable, please paste the log output in DEBUG level

See below

**Comment by hood** [comment]
Could you please provide a simple repro? I’d like to tackle this.

**Comment by guyguy333** [comment]
> Could you please provide a simple repro? I’d like to tackle this.

Hi @hood,

Thanks for checking this issue. Yes sure. I did simplify code. You can reproduce with this docker-compose. If you try to open `traefik.localhost` you should get the error (and so white page).

Please note that removing ` '--metrics.prometheus=true'` arg does not reproduce the issue.

```yaml
services:

  traefik:
    image: "traefik:v3.0.2"
    container_name: "traefik"
    command:
      - "--log.level=DEBUG"
      - "--api.insecure=true"
      - "--providers.docker=true"
      - "--providers.docker.exposedbydefault=false"
      - "--entryPoints.web.address=:80"
      - '--accesslog=true'
      - '--ping=true'
      - '--metrics.prometheus=true'
      
    ports:
      - "80:80"
      - "8080:8080"
    volumes:
      - "/var/run/docker.sock:/var/run/docker.sock:ro"
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.dashboard.rule=Host(`traefik.localhost`) && PathPrefix(`/`)"
      - "traefik.http.routers.dashboard.service=api@internal"
      - "traefik.http.routers.dashboard.middlewares=testErrors@docker,testForwardAuth@docker"
      - "traefik.http.middlewares.testForwardAuth.forwardauth.address=https://api.github.com/user/emails"
      - "traefik.http.middlewares.testForwardAuth.forwardauth.trustForwardHeader=true"
      - "traefik.http.middlewares.testForwardAuth.forwardauth.authResponseHeaders=Authorization"
      - "traefik.http.middlewares.testErrors.errors.status=401-403"
      - "traefik.http.middlewares.testErrors.errors.service=whoami"
      - "traefik.http.middlewares.testErrors.errors.query=/?rd={url}"

  whoami:
    image: "traefik/whoami"
    container_name: "whoami"
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.whoami.rule=Host(`whoami.localhost`)"
      - "traefik.http.routers.whoami.entrypoints=web"
      - "traefik.http.services.whoami.loadbalancer.server.port=80"
```

**Comment by zorgzerg** [comment]
HI guys.
I have the same problem. was there any workaround?

**Comment by georglauterbach** [comment]
I see the same problem in my logs.

**Comment by huttotw** [comment]
Try setting `metrics.addInternals` to true in the Helm chart.

**Comment by nmengin** [comment]
Hey folks,

Thanks for your feedback.
If any community member can help us to move forward in fixing the issue, we would love the help.

**Comment by zorgzerg** [comment]
> Try setting `metrics.addInternals` to true in the Helm chart.

thanks, it seems to work

**Comment by rtribotte** [comment]
Closed by #11058

**Comment by Maypul** [comment]
@rtribotte It seems it is not fixed with Traefik on Docker, as I might be getting the same issue. The service works few seconds after start of Traefik and then I am getting blank page and to make it work I needed to specify the service in each container.
I am on Traefik 3.1.2 using on Docker and using @huttotw solution and setting metrics.addInternals to true fixes this error for me.

**Comment by rtribotte** [comment]
Hello @Maypul, the fix hasn't been released yet, but should be in v3.1.3 soon.
